akbc2013/python/superuser-init.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+
+"""
+This is the initialization script for a legacy conference.
+
+It should only be run ONCE to kick off the conference. It can only be run by the Super User.
+
+"""
+
+## Import statements
+import argparse
+import csv
+import sys
+from openreview import *
+from subprocess import call
+
+## Handle the arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base URL")
+parser.add_argument('--overwrite', help="If set to true, overwrites existing groups")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+
+CONFERENCE = 'AKBC.ws/2013'
+COCHAIRS = CONFERENCE + "/Program_Co-Chairs"
+SPC = CONFERENCE + "/Senior_Program_Committee"
+PC = CONFERENCE + "/Program_Committee"
+
+
+## Initialize the client library with username and password
+if args.username!=None and args.password!=None:
+    openreview = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+else:
+    openreview = Client(baseurl=args.baseurl)
+
+groups = []
+overwrite = True if (args.overwrite!=None and args.overwrite.lower()=='true') else False
+def overwrite_allowed(groupid):
+    if not openreview.exists(groupid) or overwrite==True:
+        return True
+    else:
+        return False
+
+if openreview.user['id'].lower()=='openreview.net':
+
+    #########################
+    ##    SETUP GROUPS     ##
+    #########################
+
+    if overwrite_allowed(CONFERENCE):
+        akbc2013 = Group(CONFERENCE,
+            readers     = ['everyone'],
+            writers     = [CONFERENCE],
+            signatures  = ['OpenReview.net'],
+            signatories = [CONFERENCE],
+            members     = [],
+            web         = '../webfield/akbc2013_webfield.html')
+        groups.append(akbc2013)
+
+
+    if overwrite_allowed(COCHAIRS):
+        Program_Chairs = Group(COCHAIRS,
+            readers     = [CONFERENCE, COCHAIRS, SPC, PC],
+            writers     = [CONFERENCE],
+            signatures  = [CONFERENCE],
+            signatories = [COCHAIRS],
+            members     = [])
+        groups.append(Program_Chairs)
+
+
+    if overwrite_allowed(SPC):
+        spc = Group(SPC,
+            readers     = [CONFERENCE, COCHAIRS, SPC, PC],
+            writers     = [CONFERENCE], #the conference needs to be a writer whenever the process functions need to modify the group
+            signatures  = [CONFERENCE],
+            signatories = [SPC],
+            members     = [])
+        groups.append(spc)
+
+    if overwrite_allowed(PC):
+        pc = Group(PC,
+            readers     = [CONFERENCE, COCHAIRS, SPC, PC],
+            writers     = [CONFERENCE],
+            signatures  = [CONFERENCE],
+            signatories = [PC],
+            members     = [])
+        groups.append(pc)
+
```

Comment:
I thought they were needed, but tested without them and it works. I'll remove them in the next commit.